src/lib.rs

```diff
@@ -79,9 +81,17 @@ pub fn encrypt(
         })?);
     }
 
-    let ciphertext = encrypt_and_generate_constraints(
+    let mut ciphertext_circuit: Vec<UInt8Gadget> = Vec::with_capacity(ciphertext.len());
+    for byte in ciphertext {
+        ciphertext_circuit.push(UInt8Gadget::new_input(constraint_system.clone(), || {
+            Ok(byte)
+        })?);
+    }
```

Comment:
We could do this here.
```suggestion
    let ciphertext_circuit = UInt8Gadget::new_input_vec(constraint_system, ciphertext);
```